sdk/tdf.go

```diff
@@ -588,7 +576,7 @@ func (r *Reader) WriteTo(writer io.Writer) (int64, error) {
 		}
 
 		if int64(len(readBuf)) != seg.EncryptedSize {
-			return totalBytes, errTDFReaderFailed
+			return totalBytes, ErrTDFReaderFailed
```

Comment:
I think maybe we need an encrypted segment size mismatch here and in a few other places.